app/models/course/assessment/submission.rb

```diff
@@ -182,13 +182,29 @@ def grade
     current_answers.map { |a| a.grade || 0 }.sum
   end
 
-  # The answers with current_answer flag set to true.
+  def questions
+    assessment.randomization.nil? ? assessment.questions : assigned_questions
+  end
+
+  # The assigned questions for this submission, ordered by question_group and question_bundle_question
+  def assigned_questions
+    Course::Assessment::Question.
```

Comment:
Should cache the query result here since it is called many times.

Reply:
It's transparently cached by AR since it's always the same query:

![image](https://user-images.githubusercontent.com/11096034/53159693-5460ad80-3601-11e9-9d73-e978683ddd1b.png)